openupgrade_scripts/scripts/sale_loyalty/16.0.1.0/pre-migration.py

```diff
@@ -0,0 +1,297 @@
+        env.cr,
+        """
+        CREATE TABLE IF NOT EXISTS sale_order_coupon_points (
+            coupon_id INT,
+            order_id INT,
+            points FLOAT,
+            create_uid INT,
+            write_uid INT,
+            create_date DATE,
+            write_date DATE
+        )
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        INSERT INTO sale_order_coupon_points (
+            coupon_id, order_id, points, create_uid, write_uid, create_date, write_date
+        )
+        SELECT
+            id AS coupon_id,
+            order_id,
+            1 AS points,
+            create_uid,
+            write_uid,
+            create_date,
+            write_date
+        FROM loyalty_card;
+        """,
+    )
+
+
+def update_loyalty_program_data(env):
+    # Set sale_ok default values
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE loyalty_program
+        ADD COLUMN IF NOT EXISTS sale_ok BOOLEAN
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE loyalty_program
+        SET sale_ok = true
+        """,
+    )
+
+
+def update_sale_order_line_data(env):
+    # Fill reward_id field
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE sale_order_line
+        ADD COLUMN IF NOT EXISTS reward_id INT
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE sale_order_line sol
+        SET reward_id = lr.id
+        FROM loyalty_reward lr
+        WHERE sol.product_id = lr.discount_line_product_id
+        """,
+    )
+    # Fill coupon_id field(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE sale_order_line
+        ADD COLUMN IF NOT EXISTS coupon_id INT
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE sale_order_line
+        SET coupon_id = loyalty_card.id
+        FROM loyalty_card
+        WHERE sale_order_line.order_id = loyalty_card.sales_order_id
+        """,
+    )
+    # Fill reward_identifier_code
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE sale_order_line
+        ADD COLUMN IF NOT EXISTS reward_identifier_code CHAR
+        """,
+    )
+    reward_code = _generate_random_reward_code()
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE sale_order_line
+        SET reward_identifier_code = %s
+        WHERE reward_id IS NOT NULL
+        """,
+        (reward_code,),
+    )
+    # Fill points_cost field
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE sale_order_line
+        ADD COLUMN IF NOT EXISTS points_cost FLOAT
+        """,
+    )
+    openupgrade.logged_query(
+        openupgrade.logged_query(
+            env.cr,
+            """
+            UPDATE sale_order_line
+            SET point_cost = ABS(untaxed_amount_to_invoice)
```

Comment:
Should be points_cost
```suggestion
            SET points_cost = ABS(untaxed_amount_to_invoice)
```